ui/app/components/app/signature-request-original/signature-request-original.component.js

```diff
@@ -0,0 +1,325 @@
+    return (
+      <div className="request-signature__typed-container">
+        {
+          domain
+            ? <div>
+              <h1>
+                { 'Domain' }
+              </h1>
+              <ObjectInspector data={domain} expandLevel={1} name={'domain'} />
+            </div>
+            : ''
+        }
+        {
+          message
+            ? <div>
+              <h1>
+                {'Message'}
+              </h1>
+              <ObjectInspector data={message} expandLevel={1} name={'message'} />
+            </div>
+            : ''
+        }
+      </div>
+    )
+  }
+
+  renderBody = () => {
+    let rows
+    let notice = this.context.t('youSign') + ':'
+
+    const { txData } = this.props
+    const { type, msgParams: { data } } = txData
+
+    if (type === 'personal_sign') {
+      rows = [{ name: this.context.t('message'), value: this.msgHexToText(data) }]
+    } else if (type === 'eth_signTypedData') {
+      rows = data
+    } else if (type === 'eth_sign') {
+      rows = [{ name: this.context.t('message'), value: data }]
+      notice = (
+        <React.Fragment>
+          { this.context.t('signNotice') }
+          <span
+            className="request-signature__help-link"
+            onClick={() => {
+              global.platform.openWindow({
+                url: 'https://metamask.zendesk.com/hc/en-us/articles/360015488751',
+              })
+            }}
+          >
+            { this.context.t('learnMore') }
+          </span>
+        </React.Fragment>
+      )
+    }
+
+    return (
+      <div className="request-signature__body">
+        { this.renderAccountInfo() }
+        { this.renderRequestInfo() }
+        <div
+          className={classnames({
+            'request-signature__notice': type === 'personal_sign' || type === 'eth_signTypedData',
+            'request-signature__warning': type === 'eth_sign',
+          })}
+        >
+          { notice }
+        </div>
+        <div className="request-signature__rows">
+          {
+            rows.map(({ name, value }, index) => {
+              if (typeof value === 'boolean') {
+                value = value.toString()
+              }
+              return (
+                <div className="request-signature__row" key={`request-signature-row-${index}`}>
+                  <div className="request-signature__row-title">
+                    { `${name}:` }
+                  </div>
+                  <div className="request-signature__row-value">
+                    { value }
+                  </div>
+                </div>
+              )
+            })
+          }
+        </div>
+      </div>
+    )
+  }
+
+  renderFooter = () => {
+    const { cancel, sign } = this.props
+
+    return (
+      <div className="request-signature__footer">
+        <Button
+          type="default"
+          large
+          className="request-signature__footer__cancel-button"
+          onClick={event => {
```

Comment:
Can we make this handler `async` and then await the completion of `cancel(event)` below?